src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,122 @@
+package mate.academy.dao;
+
+import java.math.BigDecimal;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.exceptions.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.models.Book;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    @Override
+    public Book create(Book book) {
+        String sql = "INSERT INTO books (title, price) VALUE (?, ?)";
+        try (PreparedStatement prepareStatement = ConnectionUtilImpl.getConnection()
+                     .prepareStatement(sql, Statement.RETURN_GENERATED_KEYS)) {
+
+            prepareStatement.setString(1, book.getTitle());
+            prepareStatement.setBigDecimal(2, book.getPrice());
+
+            if (prepareStatement.executeUpdate() > 1) {
+                book.setId(getIdFromResultSet(prepareStatement.getGeneratedKeys()));
+                return book;
+            }
+            throw new DataProcessingException("Can't save the new book: " + book);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't save a book: " + book, e);
+        }
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String sql = "SELECT * FROM books WHERE id = ?";
+        try (PreparedStatement prepareStatement = ConnectionUtilImpl.getConnection()
+                .prepareStatement(sql)) {
+
+            prepareStatement.setLong(1, id);
+
+            return getBookFromResultSet(prepareStatement.executeQuery());
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get a book by id: " + id, e);
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        String sql = "SELECT * FROM books";
+        try (PreparedStatement prepareStatement = ConnectionUtilImpl.getConnection()
```

Comment:
```suggestion
        try (Statement prepareStatement = ConnectionUtilImpl.getConnection()
```
It's more efficient to use Statement, if there are no ?-parameters